analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/InvocationCheck.cs

```diff
@@ -0,0 +1,76 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.Checks
+{
+    internal class InvocationCheck : SymbolicCheck
+    {
+        public override ProgramState[] PreProcess(SymbolicContext context) =>
+            context.Operation.Instance.Kind == OperationKindEx.Invocation
+                && PreProcessInvocation(context.State, IInvocationOperationWrapper.FromOperation(context.Operation.Instance)) is { } newState
+                    ? newState
+                    : base.PreProcess(context);
+
+        private ProgramState[] PreProcessInvocation(ProgramState state, IInvocationOperationWrapper invocation)
```

Comment:
I find the `Invocation` in the name unnecessary. Everything in this class is about invocations. And this is still high-level orchestration method
```suggestion
        private ProgramState[] PreProcess(ProgramState state, IInvocationOperationWrapper invocation)
```